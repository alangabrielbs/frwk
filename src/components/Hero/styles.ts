import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.section`
  ${({ theme }) => css`
    width: 100%;
    max-width: 67rem;
    padding: ${theme.spacings.xxxlarge} 0;
    background: ${theme.colors.backgroundDark};
  `}
`

export const Title = styled.h1`
  ${({ theme }) => css`
    font-weight: ${theme.font.weight.bold};
    font-size: ${theme.font.heading.large};
    line-height: 5.2rem;
    background-image: ${theme.colors.gradients.primary};
    color: transparent;
    -webkit-background-clip: text;
    background-clip: text;

    ${media.greaterThan('medium')`
      font-size: ${theme.font.heading.xxxlarge};
      line-height: 8.5rem;
    `}
  `}
`

export const Description = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.content.large};
    line-height: 2.1rem;
    margin-top: ${theme.spacings.xsmall};

    color: ${theme.colors.secondaryText};

    ${media.greaterThan('medium')`
      margin-top: ${theme.spacings.small};
    `}
  `}
`
