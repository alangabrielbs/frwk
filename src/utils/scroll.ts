export const scrollTop = (top = 0) => {
  window.scrollTo({ top: top, behavior: 'smooth' })
}
