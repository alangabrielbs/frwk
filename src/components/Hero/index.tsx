import * as S from './styles'

export type HeroProps = {
  title: string
  description: string
}

const Hero = ({ title, description }: HeroProps) => (
  <S.Wrapper>
    <S.Title>{title}</S.Title>
    <S.Description>{description}</S.Description>
  </S.Wrapper>
)

export default Hero
