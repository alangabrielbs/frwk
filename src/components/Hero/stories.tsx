import { Story, Meta } from '@storybook/react'

import Hero, { HeroProps } from '.'

export default {
  title: 'Components/Hero',
  component: Hero
} as Meta

export const Default: Story<HeroProps> = (args) => <Hero {...args} />

Default.args = {
  title: 'Thoughts, Notes & Tutorials',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut pharetra hendrerit placerat. Nunc dictum nunc fermentum leo blandit porta.'
}

export const Mobile: Story<HeroProps> = (args) => <Hero {...args} />

Mobile.args = {
  title: 'Thoughts, Notes & Tutorials',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut pharetra hendrerit placerat. Nunc dictum nunc fermentum leo blandit porta.'
}

Mobile.parameters = {
  layout: 'fullscreen',
  viewport: {
    defaultViewport: 'mobile'
  }
}
