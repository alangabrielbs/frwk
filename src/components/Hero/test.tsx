import { render, screen } from 'utils/test-utils'

import Hero from '.'

const props = {
  title: 'Thoughts, Notes & Tutorials',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut pharetra hendrerit placerat. Nunc dictum nunc fermentum leo blandit porta.'
}

describe('<Hero />', () => {
  it('should render the heading', () => {
    const { container } = render(<Hero {...props} />)

    expect(
      screen.getByRole('heading', { name: props.title })
    ).toBeInTheDocument()

    expect(
      screen.getByText(props.description, { selector: 'p' })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
