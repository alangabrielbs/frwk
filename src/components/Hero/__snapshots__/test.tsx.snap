// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> should render the heading 1`] = `
.c0 {
  width: 100%;
  max-width: 67rem;
  padding: 8.0rem 0;
  background: #110D15;
}

.c1 {
  font-weight: 700;
  font-size: 4.0rem;
  line-height: 5.2rem;
  background-image: linear-gradient(110.3deg,#F85D92 8.1%,#6B81FA 91.09%);
  color: transparent;
  -webkit-background-clip: text;
  background-clip: text;
}

.c2 {
  font-size: 1.8rem;
  line-height: 2.1rem;
  margin-top: 1.6rem;
  color: #A8A8B3;
}

@media (min-width:768px) {
  .c1 {
    font-size: 7.2rem;
    line-height: 8.5rem;
  }
}

@media (min-width:768px) {
  .c2 {
    margin-top: 2.4rem;
  }
}

<section
  class="c0"
>
  <h1
    class="c1"
  >
    Thoughts, Notes & Tutorials
  </h1>
  <p
    class="c2"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut pharetra hendrerit placerat. Nunc dictum nunc fermentum leo blandit porta.
  </p>
</section>
`
